### PR TITLE
Fauna Setup

### PR DESCRIPTION
**Pull request by seanconnollydev**
Adds `fauna-gql-upload` scaffolding along with a basic frontend example to demonstrate a possible usage pattern.

The proposed structure here is to have the frontend interface with the Fauna GraphQL API directly through generated React hooks. I believe this model should cover most use cases.

### CODE REVIEW

File: components/demo/UserInfo.graphql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+query FindUserByAddress($address: String!) {
```
Comment (seanconnollydev) [inline components/demo/UserInfo.graphql:1]: Sample client-side GraphQL query definition. We have some options here as far as how we want to manage these. With small projects, I've done a single `.graphql` file and it works just fine. I think for a team project like this I would go with co-located files, i.e. if you have a React component called `UserInfo` then you'd have a co-located GQL file called `UserInfo.graphql`, which is what is demonstrated here.

File: components/demo/UserInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
```
Comment (seanconnollydev) [inline components/demo/UserInfo.tsx:1]: This demonstrates how we can use GraphQL codegen to achieve type safety in React components. This `useFindUserByAddressLazyQuery` query is autogenerated by `fauna-gql-upload` and is typesafe based on the query defined in `UserInfo.graphql`.

For example, the `findUserByAddress._id` is autocompleted and type checked, so `findUserByAddress.foo` would result in a TypeScript compilation error.

File: fauna/roles/public.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { RoleResource } from "fauna-gql-upload";
```
Comment (seanconnollydev) [inline fauna/roles/public.ts:1]: I don't think we really used roles in lvl but I think they're a good idea to embrace. I always have a public role for the entry points to the system, which in our case would be through a sign up flow. The public role will probably be limited to just user creation at some point.

File: fauna/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (seanconnollydev) [inline fauna/tsconfig.json:1]: `fauna-gql-upload` needs its own tsconfig, which kinda sucks but what can ya do.

File: generated/graphql.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
+import { gql } from '@apollo/client';
```
Comment (seanconnollydev) [inline generated/graphql.ts:1]: This file is autogenerated.

File: lib/apollo-client.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { ApolloClient, InMemoryCache } from "@apollo/client";
```
Comment (seanconnollydev) [inline lib/apollo-client.ts:1]: At some point this will need to be aware of auth, but for now it's just using the public role key.

**Review by royriver23 — APPROVED** [review]
SOLID START. This looks great, I don't have anything to call out actually. We will how the side-effects on every task can be arranged nicely (like pdf generation, discord bot, etc), but this is great for now.

**Pull request merged**